File: packages/python/examples/pytest/shadow_dom_test.py
from os import getenv

from pytest import mark


@mark.xfail(
    "appium" in getenv("ALUMNIUM_DRIVER", "selenium"),
    reason="Shadow DOM support is not implemented in Appium yet",
)
def test_shadow_dom(al, navigate):
    navigate("shadow_dom.html")

    assert "This is inside Shadow DOM!" in al.get("second paragraph")
    al.do("click first shadow button")
    assert "Shadow Button 1 was clicked!" in al.get("second paragraph")

    assert "This is another text inside Shadow DOM!" in al.get("third paragraph")
    al.do("click second shadow button")
    assert "Shadow Button 2 was clicked!" in al.get("third paragraph")

File: packages/java/src/test/java/ai/alumnium/system/ShadowDomTest.java
package ai.alumnium.system;

import java.io.File;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.condition.DisabledIfEnvironmentVariable;

@DisabledIfEnvironmentVariable(named = "ALUMNIUM_DRIVER", matches = "appium.*")
public class ShadowDomTest extends BaseTest {

  private static final String SHADOW_DOM_URL =
      new File("../python/examples/support/pages/shadow_dom.html").toURI().toString();

  @Test
  void testShadowDom() {
    navigate(SHADOW_DOM_URL);

    Assertions.assertTrue(
        ((String) al.get("second paragraph")).contains("This is inside Shadow DOM!"));
    al.act("click first shadow button");
    Assertions.assertTrue(
        ((String) al.get("second paragraph")).contains("Shadow Button 1 was clicked!"));

    Assertions.assertTrue(
        ((String) al.get("third paragraph")).contains("This is another text inside Shadow DOM!"));
    al.act("click second shadow button");
    Assertions.assertTrue(
        ((String) al.get("third paragraph")).contains("Shadow Button 2 was clicked!"));
  }
}
